### ISSUE TITLE
No image preview support for WezTerm

### ISSUE THREAD

**Issue by rice7th** [ranger/ranger#2758]
#### Runtime Environment

- Operating system and version: Arch Linux x86_64
- Terminal emulator and version: WezTerm 20221122-154621-68c754b1
- Python version: Python 3.10.8
- Ranger version/commit: ranger 1.9.3
- Locale: US_en

#### Current Behavior

Even if wezterm supports BOTH iTerm2 image protocol AND Kitty image protocol, no image is shown.

#### Expected Behavior

Show the preview of images

#### Context

I am trying to see previews of images from my terminal

#### Possible Solutions
Treat wezterm as Kitty/iTerm2 (I've heard that Kitty image protocol is more feature rich so you could use that)

#### Steps to reproduce
1. Open Ranger in WezTerm
2. Try to preview an image

**Comment by mbaz** [comment]
For me, `kitty` does not work on `wezterm` but `iterm2` does. My `rc.conf`:
```
set preview_images true
set preview_images_method iterm2
```
and
```
$ python --version
Python 3.10.8
$ ranger --version
ranger version: ranger-master
Python version: 3.10.8 (main, Nov  1 2022, 14:18:21) [GCC 12.2.0]
Locale: en_US.UTF-8
$ wezterm --version
wezterm 20221119-145034-49b9839f
```

**Comment by markus-bauer** [comment]
Probably related to this:
https://github.com/ranger/ranger/blob/f8b304f74be20c56f8596e6ea312c7c041d0def4/ranger/ext/img_display.py#L569

This attempts to check if the terminal supports the kitty protocol by querying $TERM.

Try it out, by settting the kitty preview and preview an image. This error message should show up in your message log (press W):
```'kitty previews only work in kitty and outside tmux. Make sure your TERM contains the string "kitty"'```.

I want to highlight a better implementation, by broot, that includes checks for wezterm, and querying directly:
https://github.com/Canop/broot/blob/4641155e8003e125a2d60976203901a358316789/src/kitty/detect_support.rs#L14

**Comment by mbaz** [comment]
Good catch. If I set `TERM` to `kitty` and `set preview_images_method kitty`, I get an image preview, but then ranger freezes (not surprisingly, since I'm actually running `wezterm`).

**Comment by markus-bauer** [comment]
The reason it freezes might be, because it waits for a response:
https://github.com/ranger/ranger/blob/f8b304f74be20c56f8596e6ea312c7c041d0def4/ranger/ext/img_display.py#L678

For testing, you could try it with this part commented out. 

There seem to be other issues with querying for response. I can't even get the images to render, probably because of this:
https://github.com/ranger/ranger/blob/f8b304f74be20c56f8596e6ea312c7c041d0def4/ranger/ext/img_display.py#L579

So if this is an issue, you have to handle this (for testing, I just commented it out) and set the preview method to stream = True (meaning transfer image data with the escape codes).

Anyway, I don't know what is a ranger issue and what is a wezterm issue.

I appreciate the effort by the original author, but there are so many issues with the current kitty image preview.

**Comment by toonn** [comment]
Thanks for the pointer to the broot sources. Ideally we'd do something more similar to that.

**Comment by BLumia** [comment]
Using `set preview_images_method iterm2` works well in my case, no `TERM` env needs to be exported, the only issue is the preview size since it depends on `iterm2_font_height`, and can be overwrite in config file. e.g. https://github.com/ranger/ranger/issues/2254#issuecomment-1782547277
